plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/worker/KtLintClassesSerializer.kt

```diff
@@ -19,28 +17,17 @@ internal interface KtLintClassesSerializer {
         serializedErrors: File
     ): List<LintErrorResult>
 
-    fun saveReporterProviders(
-        reporterProviders: List<ReporterProvider<*>>,
+    fun <T : Serializable> saveReporterProviders(
+        reporterProviders: List<T>,
         serializedReporterProviders: File
     )
 
-    fun loadReporterProviders(
-        serializedReporterProviders: File
-    ): List<ReporterProvider<*>>
-
     companion object {
-        fun create(ktLintVersion: SemVer): KtLintClassesSerializer =
-            if (ktLintVersion >= SemVer(0, 41, 0)) {
-                CurrentKtLintClassesSerializer()
-            } else {
-                OldKtLintClassesSerializer()
-            }
+        fun create(ktLintVersion: SemVer? = null): KtLintClassesSerializer =
```

Comment:
This argument doesn't look like it's used. Do we need it?

Reply:
fixed. thanks!